packages/ButtonGroup/src/ButtonGroup.js

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+import PropTypes from "prop-types";
+
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+
+import * as sc from "./ButtonGroup.styles";
+
+const propTypes = {
+  /** The content of each of the toggle buttons in the group. */
+  children: PropTypes.node,
+
+  /** T show or hide the icons for selected or not. */
+  hasIcons: PropTypes.bool,
+
+  /** If the button is disabled. */
+  isDisabled: PropTypes.bool,
+
+  /** If the width of the button should span it's parent container (100%). */
+  isFullWidth: PropTypes.bool,
+
+  /** If it will be rendered as a <button> element. If false, a <span> will be rendered via an accessible <RawButton>. */
+  isSemantic: PropTypes.bool,
+
+  /** Callback to be executed when any button is clicked to select the item or activated by keyboard. Typically required.
+   * it will return a list of selected items */
+  onChange: PropTypes.func,
+
+  /** Size of the button (font size, min-height, padding, etc). */
+  size: PropTypes.oneOf(ShirtSizes.DEFAULT),
+
+  /** Value for tabindex attribute to override the default of 0. */
+  tabIndex: PropTypes.number,
+};
+
+const defaultProps = {
+  children: null,
+  hasIcons: false,
+  isDisabled: false,
+  isFullWidth: false,
+  isSemantic: true,
+  onChange: () => {},
+  size: ShirtSizes.MEDIUM,
+  tabIndex: null,
+};
+
+const ButtonGroup = props => {
+  const { children, hasIcons, isDisabled, isFullWidth, isSemantic, onChange, tabIndex, size, ...moreProps } = props;
+
+  const [selectedItems, setSelectedItems] = React.useState([]);
+  const isButtonGroupDisabled = isDisabled;
+
+  const handleClick = event => {
+    if (!isButtonGroupDisabled) {
+      const clickedId = event.target.textContent;
```

Comment:
This works okay in your example where the content of each button is just a string, but theoretically you can have a `node` there, which could get weird. I wonder if it would be better to have a required `value` prop for each `ButtonItem`, kind of like a checkbox or radio control.  That would need to be unique, and could be used for the key, and for the onClick API. What do you think?

Reply:
... it could also be different than the visible label too, and more useful for the application developer that might care about some data value more than the (localized) label on the button.

Reply:
Nice idk I think it makes sense, I'm just having a hard time visualizing how can node be more than just the button with the label. I guess technically we are allowing the user to pass in anything, but maybe that is the restriction we need instead?

If we go down the path you suggest would we need to define an `onClick` for each ButtonItem to bubble up the value prop? or how would that value be captured on the event?